src/core/router/transports/ssu/packet.h

```diff
@@ -755,8 +755,11 @@ class SSUPacketParser : private kovri::core::InputByteStream {
   std::unique_ptr<SSUSessionCreatedPacket> ParseSessionCreated();
 
   /// @brief Parses a session confirmed packet, without the header
+  /// @param header Packet header
+  /// @note header is required to compute padding
   /// TODO: Support multiple fragments?
-  std::unique_ptr<SSUSessionConfirmedPacket> ParseSessionConfirmed();
+  std::unique_ptr<SSUSessionConfirmedPacket> ParseSessionConfirmed(
+      const SSUHeader* header);
```

Comment:
> Any thoughts?

In addition, imho, the only public member in SSUPacketParser should be ParsePacket. Ideally, parser should be destroyed just after return the newly created packet (since the header moves from the parser to packet). 
Likewise, SSUPacketBuilder::WriteSession[...] should be private or protected. 
That's why i changed a little the unit-test to use ParsePacket and WritePacket instead of ParseSessionConfirmed and WriteSessionConfirmed.